A32NX/html_ui/Pages/VCockpit/Instruments/Airliners/A320_Neo/FMC/A32NX_FMCMainDisplay.js

```diff
@@ -1302,6 +1302,51 @@ class FMCMainDisplay extends BaseAirliners {
         return this._routeReservedPercent;
     }
 
+    /**
+     * Sets new Cruise FL if all conditions good
+     * @param input {float} Altitude or FL
+     * @returns {boolean} input passed checks
+     */
+    trySetCruiseFl(input) {
+        if (input === FMCMainDisplay.clrValue) {
+            this.showErrorMessage(this.defaultInputErrorMessage);
+            return false;
+        }
+        const flString = input.replace("FL", "");
+        if (!flString) {
+            this.showErrorMessage(this.defaultInputErrorMessage);
+            return false;
+        }
+        let fl = parseFloat(flString);
+        if (!isFinite(fl)) {
+            this.showErrorMessage(this.defaultInputErrorMessage);
+            return false;
+        }
+        if (fl >= 1000) {
+            fl = Math.floor(fl / 100);
+        }
+        if (fl > this.maxCruiseFL) {
+            this.showErrorMessage(this.defaultInputErrorMessage);
+            return false;
+        }
+        const phase = Simplane.getCurrentFlightPhase();
+        if (this._cruiseEntered && fl < this.cruiseFlightLevel && phase === FlightPhase.FLIGHT_PHASE_CRUISE) {
```

Comment:
The reference you provided in the initial PR comment specifies:

> During CRUISE it is not possible to insert a cruise flight level on the progress page that is below the FCU altitude

This code checks against the current MCDU cruise flight level, not the FCU flight level (aka the "autopilot setting").

Reply:
you are absolutely correct, I fixed it.